src/main/java/com/monumental/models/Monument.java

```diff
@@ -258,4 +267,108 @@ private String generateDescription() {
 
         return description;
     }
+
+    /**
+     * Adds a Contribution to the List
+     * Will make a new ArrayList if this.contributions is null
+     * Checks if a Contribution has already been added to this.contributions with the same date and name
+     * Will do nothing if the specified Contribution has a null date or null/empty submittedBy
+     * and does nothing if so
+     * @param contribution - Contribution to add to the List
+     */
+    public void addContribution(Contribution contribution) {
```

Comment:
These are covered by the `NotNull` and such not aren't they? do we still need these? I think this is bad design if we do need them

Reply:
Well, asserting that they aren't null on insert/update of a Monument would be covered by `@NotNull`, but I think we still need a way to associate the Contribution with the Monument in-memory before the insert/update like we do for Tags, no?

Reply:
It seems like the entire point of these `addBlank` functions is to do a null check for you. Why not just initialize those lists in the `Monument` constructor if they're `null`?

```java
public Monument(...) {
    if (this.getContributions() == null) this.setContributions(new ArrayList<>());
    // etc.
}
```

Reply:
Fair enough you have provided all valid points. I'll fix that now.